src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -523,6 +523,44 @@ public void TestStartWithRootUser()
             RunTestAsSudo(TestStartWithUserName, GetCurrentRealUserName());
         }
 
+        /// <summary>
+        /// Tests when running as root and starting a new process as a normal user,
+        /// the new process doesn't have elevated privileges.
+        /// </summary>
+        [Fact]
+        [OuterLoop("Needs sudo access")]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestCheckChildProcessUserIds()
+        {
+            Func<string, string, string, string, int> runsAsRoot = (string username, string arg1, string arg2, string arg3) =>
+            {
+                Func<string, string, string, int> runsAsUser = (string uid, string gid, string groupsJoined) =>
+                {
+                    uint[] groups = groupsJoined.Split(',').Select(s => uint.Parse(s)).ToArray();
+
+                    Assert.Equal(uid, getuid().ToString());
+                    Assert.Equal(gid, getgid().ToString());
+                    Assert.Equal(groups, GetGroups());
+
+                    return SuccessExitCode;
+                };
+
+                var invokeOptions = new RemoteInvokeOptions();
+                invokeOptions.StartInfo.UserName = username;
+                using (RemoteInvokeHandle handle = RemoteInvoke(runsAsUser, arg1, arg2, arg3, invokeOptions))
+                { }
+
+                return SuccessExitCode;
+            };
+
+            string currentUid = getuid().ToString();
+            string currentGid = getgid().ToString();
+            string currentGroups = string.Join(",", GetGroups());
+            using (RemoteInvokeHandle handle = RemoteInvoke(runsAsRoot, GetCurrentRealUserName(), currentUid, currentGid, currentGroups,
+                                                            new RemoteInvokeOptions { RunAsSudo = true }))
+            { }
+        }
```

Comment:
It would be nice to test running as a user that's not self, since that exercises the fix. Perhaps, if sudoable, and outerloop, it could create a new user (named with some guid) and use that (removing it afterwards), verifying it couldn't write where it shouldn't.

Reply:
We become root in-between. I will add some asserts that verify we actually became root.

Reply:
Ah. Is the case regularuser1 -> regularuser2 important?

Reply:
The test now cover two cases:
regularuser1 -> regularuser1
privileged user (root) -> regularuser1